packages/hash/subgraph/src/stdlib/element/entity.ts

```diff
@@ -0,0 +1,135 @@
+import { Subgraph } from "../../types/subgraph";
+import {
+  EntityEditionId,
+  EntityId,
+  isEntityEditionId,
+} from "../../types/identifier";
+import { isEntityVertex } from "../../types/vertex";
+import { Entity } from "../../types/element";
+import { mustBeDefined } from "../../shared/invariant";
+
+/**
+ * Returns all `Entity`s within the vertices of the subgraph
+ *
+ * @param subgraph
+ */
+export const getEntities = (subgraph: Subgraph): Entity[] => {
+  return Object.values(
+    Object.values(subgraph.vertices).flatMap((versionObject) =>
+      Object.values(versionObject)
+        .filter(isEntityVertex)
+        .map((vertex) => vertex.inner),
+    ),
+  );
+};
+
+/**
+ * Gets an `Entity` by its `EntityEditionId` from within the vertices of the subgraph. Returns `undefined`
+ * if the entity couldn't be found.
+ *
+ * @param subgraph
+ * @param entityEditionId
+ * @throws if the vertex isn't an `EntityVertex`
+ */
+export const getEntityByEditionId = (
+  subgraph: Subgraph,
+  entityEditionId: EntityEditionId,
+): Entity | undefined => {
+  const [entityId, version] = [entityEditionId.baseId, entityEditionId.version];
```

Comment:
Nit: similar to a prior nit, I think de-structuring the object is more appropriate here and be simpler to read:
```suggestion
  const { baseId: entityId, version } = entityEditionId;
```

Reply:
Yep this one is objectively better, I'll update, thanks!